algorithm/eight_queens.py

```diff
@@ -29,15 +34,185 @@ def board(self) -> list[list[dict]]:
+            i += 1
+
+        # quadrant 2 (upper left)
+        i=1
+        while row-i>=0 and column-i>=0:
+            self._board[row-i][column-i]["value"] += 1
+
+            if tile_is_empty(row-i, column-i):
+                self._board[row-i][column-i]["id"] = "killzone"    
+                self._board[row-i][column-i]["asset"] = self._killzone_asset
+            i += 1
+
+        # quadrant 3 (bottom right)
+        i=1
+        while row+i<self.size and column-i>=0:
+            self._board[row+i][column-i]["value"] += 1
+
+            if tile_is_empty(row+i, column-i):
+                self._board[row+i][column-i]["id"] = "killzone"
+                self._board[row+i][column-i]["asset"] = self._killzone_asset
+            i += 1
+
+        # quadrant 4 (bottom left)
+        i=1
+        while row+i<self.size and column+i<self.size:
+            self._board[row+i][column+i]["value"] += 1
+
+            if tile_is_empty(row+i, column+i):
+                self._board[row+i][column+i]["id"] = "killzone"
+                self._board[row+i][column+i]["asset"] = self._killzone_asset
+            i += 1
+
+        self._validate_queens()
 
     def remove_queen(self, position: tuple):
+        """
+        Removes a queen and their respective killzones on the board.
+        Removed killzones that overlapped with other killzones won't remove those other killzones
+
+        Args:
+            position (tuple): row and column of the queen to be removed.
+        Raises:
+            ValueError: If the specified position is not occupied by a queen.
+        """
         self._validate_position(position)
+        
+        occupied = self._board[row][column]["id"] == "queen"
+        if not occupied:
+            EMPTY_TILE = f"This tile ({row},{column}) does not have a queen. It has {self._board[row][column]['id']} as the id\nDid you mean to use 'place_queen()' instead?"
+            raise ValueError(EMPTY_TILE)
+        
+        self._queen_positions.pop(self._queen_positions.index(position))
+
+        row, column = position
+        self._board[row][column]["id"] = None
+
+        removed_queen_was_killzone = self._board[row][column]["value"] > 0
+        if removed_queen_was_killzone:
+            self._board[row][column]["id"] = "killzone"
+
+        def removed_killzone_is_empty(row: int, column: int) -> bool:
+            'helper subfunction to avoid replacing asset of an invalid queen when removing a killzone'
+            return self._board[row][column]["value"] == 0 and self._board[row][column]["asset"] == self._killzone_asset
+
+        # row and column killzone
+        for i in range(self.size):
+            # rows
+            removed_queen = column
+            if i != removed_queen:
+                self._board[row][i]["value"] -= 1
+
+                if removed_killzone_is_empty(row, i):
+                    self._board[row][i]["id"] = None
+                    self._board[row][i]["asset"] = None
```

Comment:
same comment from place queen, code snippet is repeated multiple times, might be better to create a sub fn for it
```suggestion
        def remove_killzone(coord,i):
                if is_removed_killzone_empty(coord, i):
                    self._board[coord][i]["id"] = None
                    self._board[coord][i]["asset"] = None
```